great_expectations/experimental/datasources/experimental_base_model.py

```diff
@@ -37,6 +37,19 @@
 
 
 class ExperimentalBaseModel(pydantic.BaseModel):
+    """
+    Base model for most ZEP pydantic models.
+
+    Adds yaml dumping and parsing methods.
+
+    Extra fields are not allowed.
+
+    Serialization methods default to `exclude_unset = True` to prevent serializing
+    configs full of mostly unset default values.
+    Also prevents passing along unset kwargs to BatchSpec.
+    https://docs.pydantic.dev/usage/exporting_models/
```

Comment:
Making this the default for all of our models does not work.
It leads to weird behavior where a `Datasource` doesn't serialize its own assets.

I believe this is due to things that are being set in `pre` validators not registering as being "set", which actually makes sense when thinking about using `pre` validators to set dynamic default values like in the example below.
https://docs.pydantic.dev/usage/validators/#validate-always

There are ways I could make this work but I believe the simpler method is to make this the default for all `_DataFrameAsset`s and keep the rest of the models alone.

Reply:
It does work, we just need to make use of the pydantic `__fields_set__` attribute if we add items to a model post-init.